File: .codex/skills/project-scaffolding/assets/nestjs-prisma-postgres/src/messages/messages.controller.ts
import { Body, Controller, Post, Req, UseGuards } from "@nestjs/common";
import { PrismaService } from "../prisma.service";
import { AuthGuard } from "../auth/auth.guard";

@Controller("messages")
export class MessagesController {
  constructor(private readonly prisma: PrismaService) {}

  @UseGuards(AuthGuard)
  @Post("send")
  async send(
    @Req() req: { user: { sub: string } },
    @Body() body: { payload: string; channel?: string }
  ) {
    const message = await this.prisma.message.create({
      data: {
        senderId: req.user.sub,
        payload: body.payload,
        channel: body.channel || "in_app",
        status: "QUEUED",
      },
    });
    return {
      message,
      nextAction: "connect_queue_or_provider",
    };
  }
}

File: .codex/skills/project-scaffolding/assets/java-spring-postgres/src/main/java/com/codexkit/template/controller/MessagesController.java
package com.codexkit.template.controller;

import com.codexkit.template.domain.Message;
import com.codexkit.template.domain.User;
import com.codexkit.template.repo.MessageRepository;
import com.codexkit.template.service.AuthService;
import java.util.Map;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/messages")
public class MessagesController {
  private final AuthService authService;
  private final MessageRepository messageRepository;

  public MessagesController(AuthService authService, MessageRepository messageRepository) {
    this.authService = authService;
    this.messageRepository = messageRepository;
  }

  @PostMapping("/send")
  public Map<String, Object> send(
      @RequestHeader("Authorization") String authorization,
      @RequestBody Map<String, Object> payload) {
    User user = authService.requireUser(authorization);
    Message message = new Message();
    message.setSender(user);
    message.setChannel(String.valueOf(payload.getOrDefault("channel", "in_app")));
    message.setPayload(String.valueOf(payload.get("payload")));
    message.setStatus("QUEUED");
    messageRepository.save(message);
    return Map.of("message", message, "next_action", "connect_queue_or_provider");
  }
}

